### ISSUE TITLE
[BUG] WinGet troubleshooter should also delete temp folders

### ISSUE THREAD

**Issue by Ilmereth** [marticliment/UniGetUI#2881]
### Please confirm these before moving forward

- [X] I have searched for my issue and have not found a work-in-progress/duplicate/resolved issue.
- [X] I have tested that this issue has not been fixed in the latest [(beta or stable) release](https://github.com/marticliment/WingetUI/releases/).
- [X] I have checked the [FAQ](https://github.com/marticliment/WingetUI#frequently-asked-questions) section for solutions.
- [X] This issue is about a bug (if it is not, please use the [correct template](https://github.com/marticliment/WingetUI/issues/new/choose)).

### UniGetUI Version

3.1.2

### Windows version, edition, and architecture

Windows 11 Pro 23H2

### Describe your issue

I installed UniGetUI from the Ddirect Download Link on https://www.marticliment.com/unigetui/.
When I started it, I got a yellowish bar at the top of the UniGetUI telling me:
WinGet malfunction detected. It looks like WinGet is not working properly. Do you want to attempt to repair WinGet?
with a Button Repair WinGet on the right corner.
Whe I click this, a PowerShell Window opens installing microsoft.winget.client  from nuget, after telling me: 

FEHLER: Der Prozess "winget.exe" wurde nicht gefunden.
FEHLER: Der Prozess "WindowsPackageManagerServer.exe" wurde nicht gefunden.

I tried all versions listed on https://github.com/marticliment/UniGetUI/releases/tag/3.1.2
as a portable install as well as a normal install and could reproduce the error in all installs.

### Steps to reproduce the issue

Click on RepairWinGet Button -> Restart -> Get Error: WinGet malfunction detected -> Repeat

### UniGetUI Log

```text
[23.10.2024 11:20:27]    __  __      _ ______     __  __  ______
                        / / / /___  (_) ____/__  / /_/ / / /  _/
                       / / / / __ \/ / / __/ _ \/ __/ / / // /
                      / /_/ / / / / / /_/ /  __/ /_/ /_/ // /
                      \____/_/ /_/_/\____/\___/\__/\____/___/
                          Welcome to UniGetUI Version 3.1.2
[23.10.2024 11:20:27]
[23.10.2024 11:20:27] Version Code:  3,12
[23.10.2024 11:20:27] Encoding Code Page set to 850
[23.10.2024 11:20:27] Using bundled GSudo at C:\Program Files\UniGetUI\Assets\Utilities\gsudo.exe
[23.10.2024 11:20:27] Loaded language locale: en
[23.10.2024 11:20:27] Starting update check
[23.10.2024 11:20:27] Randomly-generated background API auth token for the current session: nm4fcgkg7qklm7i4bhun7m9ii6hgrrnmoarzjnsnoxa3jf6jwj94skcoynexwb4m
[23.10.2024 11:20:27] Api running on http://localhost:7058
[23.10.2024 11:20:27] Lang files were updated successfully from GitHub
[23.10.2024 11:20:27] Begin "which" search for command cargo
[23.10.2024 11:20:27] Begin "which" search for command npm
[23.10.2024 11:20:27] Begin "which" search for command dotnet.exe
[23.10.2024 11:20:27] Begin "which" search for command winget.exe
[23.10.2024 11:20:27] Begin "which" search for command pwsh.exe
[23.10.2024 11:20:27] Begin "which" search for command python.exe
[23.10.2024 11:20:27] Old chocolatey path does not exist, not migrating Chocolatey
[23.10.2024 11:20:27] Downloaded new icons and screenshots successfully!
[23.10.2024 11:20:27] Command winget.exe was found on C:\Users\matth\AppData\Local\Microsoft\WindowsApps\winget.exe
[23.10.2024 11:20:27] Command pwsh.exe was not found on the system
[23.10.2024 11:20:27] ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
                      █▀▀▀▀▀▀▀▀▀▀▀▀▀ MANAGER LOADED ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
                      █ Name: PowerShell7
                      █ Enabled: True
                      █ Found: False
                      █ THE MANAGER WAS NOT FOUND. PERHAPS IT IS NOT 
                      █ INSTALLED OR IT HAS BEEN MISCONFIGURED 
                      ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
[23.10.2024 11:20:27] Command cargo was not found on the system
[23.10.2024 11:20:27] ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
                      █▀▀▀▀▀▀▀▀▀▀▀▀▀ MANAGER LOADED ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
                      █ Name: Cargo
                      █ Enabled: True
                      █ Found: False
                      █ THE MANAGER WAS NOT FOUND. PERHAPS IT IS NOT 
                      █ INSTALLED OR IT HAS BEEN MISCONFIGURED 
                      ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
[23.10.2024 11:20:27] Command dotnet.exe was found on C:\Program Files\dotnet\dotnet.exe
[23.10.2024 11:20:27] Command python.exe was found on C:\Users\matth\AppData\Local\Microsoft\WindowsApps\python.exe
[23.10.2024 11:20:27] Command npm was not found on the system
[23.10.2024 11:20:27] ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
                      █▀▀▀▀▀▀▀▀▀▀▀▀▀ MANAGER LOADED ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
                      █ Name: Npm
                      █ Enabled: True
                      █ Found: False
                      █ THE MANAGER WAS NOT FOUND. PERHAPS IT IS NOT 
                      █ INSTALLED OR IT HAS BEEN MISCONFIGURED 
                      ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
[23.10.2024 11:20:27] ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
                      █▀▀▀▀▀▀▀▀▀▀▀▀▀ MANAGER LOADED ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
                      █ Name: .NET Tool
                      █ Enabled: True
                      █ Found: False
                      █ THE MANAGER WAS NOT FOUND. PERHAPS IT IS NOT 
                      █ INSTALLED OR IT HAS BEEN MISCONFIGURED 
                      ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
[23.10.2024 11:20:27] ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
                      █▀▀▀▀▀▀▀▀▀▀▀▀▀ MANAGER LOADED ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
                      █ Name: Pip
                      █ Enabled: True
                      █ Found: True
                      █ Fancy exe name: pip
                      █ Executable path: C:\Users\matth\AppData\Local\Microsoft\WindowsApps\python.exe
                      █ Call arguments:  -m pip
                      █ Version: 
                      █   
                      ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
[23.10.2024 11:20:27] UniGetUI is up to date
[23.10.2024 11:20:27] Loaded 2 sources for manager Winget
[23.10.2024 11:20:27] ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
                      █▀▀▀▀▀▀▀▀▀▀▀▀▀ MANAGER LOADED ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
                      █ Name: Winget
                      █ Enabled: True
                      █ Found: True
                      █ Fancy exe name: winget.exe
                      █ Executable path: C:\Users\matth\AppData\Local\Microsoft\WindowsApps\winget.exe
                      █ Call arguments: 
                      █ Version: 
                      █   System WinGet CLI Version: v1.8.1911
                      █   Using Native WinGet helper (COM Api)
                      ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
[23.10.2024 11:20:27] Begin "which" search for command scoop
[23.10.2024 11:20:28] Command scoop was found on C:\Users\matth\scoop\shims\scoop
[23.10.2024 11:20:28] Loaded 1 sources for manager Scoop
[23.10.2024 11:20:28] ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
                      █▀▀▀▀▀▀▀▀▀▀▀▀▀ MANAGER LOADED ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
                      █ Name: Scoop
                      █ Enabled: True
                      █ Found: True
                      █ Fancy exe name: scoop
                      █ Executable path: C:\Windows\system32\windowspowershell\v1.0\powershell.exe
                      █ Call arguments:  -NoProfile -ExecutionPolicy Bypass -Command scoop
                      █ Version: 
                      █   Current Scoop version:
                      █   859d1db5 chore(release): Bump to version 0.5.2 (#6080)
                      █   
                      █   'main' bucket:
                      █   e67995ed7 lefthook: Update to version 1.8.1
                      ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
[23.10.2024 11:20:28] Loaded 1 sources for manager Chocolatey
[23.10.2024 11:20:28] ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
                      █▀▀▀▀▀▀▀▀▀▀▀▀▀ MANAGER LOADED ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
                      █ Name: Chocolatey
                      █ Enabled: True
                      █ Found: True
                      █ Fancy exe name: choco.exe
                      █ Executable path: C:\Users\matth\AppData\Local\UniGetUI\Chocolatey\choco.exe
                      █ Call arguments: 
                      █ Version: 
                      █   2.2.2
                      ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
[23.10.2024 11:20:29] Loaded 1 sources for manager PowerShell
[23.10.2024 11:20:29] ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
                      █▀▀▀▀▀▀▀▀▀▀▀▀▀ MANAGER LOADED ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
                      █ Name: PowerShell
                      █ Enabled: True
                      █ Found: True
                      █ Fancy exe name: powershell.exe
                      █ Executable path: C:\Windows\system32\windowspowershell\v1.0\powershell.exe
                      █ Call arguments:  -NoProfile -Command
                      █ Version: 
                      █   Name                           Value                                                                                   
                      █   ----                           -----                                                                                   
                      █   PSVersion                      5.1.22621.4249                                                                          
                      █   PSEdition                      Desktop                                                                                 
                      █   PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                 
                      █   BuildVersion                   10.0.22621.4249                                                                         
                      █   CLRVersion                     4.0.30319.42000                                                                         
                      █   WSManStackVersion              3.0                                                                                     
                      █   PSRemotingProtocolVersion      2.3                                                                                     
                      █   SerializationVersion           1.1.0.1
                      ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
[23.10.2024 11:20:29] LoadComponentsAsync finished executing. All managers loaded. Proceeding to interface.
[23.10.2024 11:20:29] Manager Pip has not implemented RefreshPackageIndexes
[23.10.2024 11:20:29] Manager PowerShell has not implemented RefreshPackageIndexes
[23.10.2024 11:20:29] Manager Chocolatey has not implemented RefreshPackageIndexes
[23.10.2024 11:20:29] Found 0 available updates from Pip
[23.10.2024 11:20:29] Found 0 installed packages from Pip
[23.10.2024 11:20:29] Loaded 2 sources for manager Winget
[23.10.2024 11:20:29] Begin "which" search for command scoop-search.exe
[23.10.2024 11:20:29] Command scoop-search.exe was found on C:\Users\matth\scoop\shims\scoop-search.exe
[23.10.2024 11:20:29] Dependency Scoop-Search for manager Scoop is present
[23.10.2024 11:20:29] Begin "which" search for command git.exe
[23.10.2024 11:20:29] Command git.exe was found on C:\Users\matth\scoop\shims\git.exe
[23.10.2024 11:20:29] Dependency Git for manager Scoop is present
[23.10.2024 11:20:29] Found 3 installed packages from Scoop
[23.10.2024 11:20:29] Ignored updates database was never loaded, so it is going to be loaded now
[23.10.2024 11:20:29] Loaded 1 sources for manager Scoop
[23.10.2024 11:20:29] Found 0 installed packages from Chocolatey
[23.10.2024 11:20:29] Found 0 available updates from Chocolatey
[23.10.2024 11:20:29] Loaded 1 sources for manager Chocolatey
[23.10.2024 11:20:30] Found 1 installed packages from PowerShell
[23.10.2024 11:20:30] Loaded 1 sources for manager PowerShell
[23.10.2024 11:20:30] Found 3 installed packages from Scoop
[23.10.2024 11:20:31] Found 0 available updates from PowerShell
[23.10.2024 11:20:31] Found 0 available updates from Scoop
[23.10.2024 11:20:36] Error finding installed packages on manager Winget
[23.10.2024 11:20:36] System.AggregateException: One or more errors occurred. (WinGet: Failed to connect to composite catalog.)
                       ---> System.InvalidOperationException: WinGet: Failed to connect to composite catalog.
                         at UniGetUI.PackageEngine.Managers.WingetManager.NativeWinGetHelper.GetLocalWinGetPackages(INativeTaskLogger logger) in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.Managers.WinGet\WinGetHelpers\NativeWinGetHelper.cs:line 211
                         at UniGetUI.PackageEngine.Managers.WingetManager.NativeWinGetHelper.GetInstalledPackages_UnSafe(WinGet Manager) in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.Managers.WinGet\WinGetHelpers\NativeWinGetHelper.cs:line 176
                         at UniGetUI.PackageEngine.Managers.WingetManager.WinGet.GetInstalledPackages_UnSafe() in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.Managers.WinGet\WinGet.cs:line 102
                         at System.Threading.Tasks.Task`1.InnerInvoke()
                         at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
                      --- End of stack trace from previous location ---
                         at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
                         at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
                         --- End of inner exception stack trace ---
                         at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
                         at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
                         at System.Threading.Tasks.Task.Wait(TimeSpan timeout, CancellationToken cancellationToken)
                         at System.Threading.Tasks.Task.Wait(TimeSpan timeout)
                         at UniGetUI.PackageEngine.ManagerClasses.Manager.PackageManager.GetInstalledPackages() in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.PackageManagerClasses\Manager\PackageManager.cs:line 228
[23.10.2024 11:20:41] Error finding updates on manager Winget
[23.10.2024 11:20:41] System.AggregateException: One or more errors occurred. (WinGet: Failed to connect to composite catalog.)
                       ---> System.InvalidOperationException: WinGet: Failed to connect to composite catalog.
                         at UniGetUI.PackageEngine.Managers.WingetManager.NativeWinGetHelper.GetLocalWinGetPackages(INativeTaskLogger logger) in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.Managers.WinGet\WinGetHelpers\NativeWinGetHelper.cs:line 211
                         at UniGetUI.PackageEngine.Managers.WingetManager.NativeWinGetHelper.GetAvailableUpdates_UnSafe(WinGet Manager) in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.Managers.WinGet\WinGetHelpers\NativeWinGetHelper.cs:line 156
                         at UniGetUI.PackageEngine.Managers.WingetManager.WinGet.GetAvailableUpdates_UnSafe() in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.Managers.WinGet\WinGet.cs:line 95
                         at System.Threading.Tasks.Task`1.InnerInvoke()
                         at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
                      --- End of stack trace from previous location ---
                         at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
                         at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
                         --- End of inner exception stack trace ---
                         at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
                         at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
                         at System.Threading.Tasks.Task.Wait(TimeSpan timeout, CancellationToken cancellationToken)
                         at System.Threading.Tasks.Task.Wait(TimeSpan timeout)
                         at UniGetUI.PackageEngine.ManagerClasses.Manager.PackageManager.GetAvailableUpdates() in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.PackageManagerClasses\Manager\PackageManager.cs:line 197
[23.10.2024 11:20:41] Invalid value for UpdatesCheckInterval, using default value of 3600 seconds
```

### Package Managers Logs

```text
Manager WinGet with version:
System WinGet CLI Version: v1.8.1911
Using Native WinGet helper (COM Api)

——————————————————————————————————————————


Logged native task on manager Winget. Task type is ListSources
Process start time: 23.10.2024 11:25:53
Process end time:   23.10.2024 11:25:53

-- Task information
 ...

The task reported success

——————————————————————————————————————————

Logged subprocess-based task on manager Winget. Task type is RefreshIndexes
Subprocess executable: "C:\Users\matth\AppData\Local\Microsoft\WindowsApps\winget.exe"
Command-line arguments: " source update --disable-interactivity"
Process start time: 23.10.2024 11:25:54
Process end time:   23.10.2024 11:25:58

-- Process STDOUT
 ...

Return code: SUCCESS (0)

——————————————————————————————————————————

Logged native task on manager Winget. Task type is ListInstalledPackages
Process start time: 23.10.2024 11:25:54
Process end time:   23.10.2024 11:25:57

-- Task information
 ...

-- Task errors
  Failed to connect to installedSearchCatalogRef. Aborting.

The task reported a failure (1)

——————————————————————————————————————————

Logged native task on manager Winget. Task type is ListSources
Process start time: 23.10.2024 11:25:54
Process end time:   23.10.2024 11:25:54

-- Task information
 ...

The task reported success

——————————————————————————————————————————

Logged native task on manager Winget. Task type is ListUpdates
Process start time: 23.10.2024 11:25:58
Process end time:   23.10.2024 11:26:06

-- Task information
 ...

-- Task errors
  Failed to connect to installedSearchCatalogRef. Aborting.

The task reported a failure (1)

——————————————————————————————————————————

Logged native task on manager Winget. Task type is ListInstalledPackages
Process start time: 23.10.2024 11:26:52
Process end time:   23.10.2024 11:26:55

-- Task information
 ...

-- Task errors
  Failed to connect to installedSearchCatalogRef. Aborting.

The task reported a failure (1)

——————————————————————————————————————————
```

**Comment by tobiulm** [comment]
The solution: https://github.com/microsoft/winget-cli/issues/2930#issuecomment-2342936733

**Comment by marticliment** [comment]
Oh, I will add the temp folder reset strategy to the script.

Thanks for pointing out

**Comment by marticliment** [comment]
This was fixed on 3.1.3
